h5netcdf/core.py

```diff
@@ -289,7 +289,7 @@ def _group_cls(self):
     def __init__(self, parent, name):
         self._parent_ref = weakref.ref(parent)
         self._root_ref = weakref.ref(parent._root)
-        self._h5path = _join_h5paths(parent.name, name)
+        self._h5path = _join_h5paths(parent._h5path, name)
```

Comment:
This always gets the full path for constructing the h5path since .name gives back group name only for legacyapi.